src/motion/motion.ts

```diff
@@ -81,8 +81,10 @@ export class Motion implements vscode.Disposable {
     }
 
     public move(line : number = null, character : number = null) : Motion {
-        if (line !== null && character !== null) {
+        if (line !== null) {
             this._position = this.position.setLocation(line, character);
+        }
+        if (character !== null && character > this._desiredColumn) {
```

Comment:
The thing is, that for me when I moved up/down it reset the _desiredColumn to 0 for empty lines.

When you move left or right by one character it already resets the _desiredColumn as far as I could tell